services/server/server.go

```diff
@@ -546,22 +546,35 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 		return nil, err
 	}
 
-	if err := s.supervisord.UpdateConfiguration(settings); err != nil {
-		return nil, err
-	}
 	if isAgentsStateUpdateNeeded(req.MetricsResolutions) {
 		if err := s.r.UpdateAgentsState(ctx); err != nil {
 			return nil, err
 		}
 	}
-	s.vmdb.RequestConfigurationUpdate()
-	s.vmalert.RequestConfigurationUpdate()
+
+	if err := s.UpdateConfigurations(); err != nil {
+		return nil, err
+	}
 
 	return &serverpb.ChangeSettingsResponse{
 		Settings: s.convertSettings(settings),
 	}, nil
 }
 
+// UpdateConfigurations updates supervisor config and requests configuration update for victoria metrics
+func (s *Server) UpdateConfigurations() error {
```

Comment:
Given that we have a separate method now, can we add a unit test?

Reply:
@Dasio ^

Reply:
I discussed it with Aleksi, and we came up, there is no good unit test for this.